components/automate-ui/src/app/pages/notification-details/notification-details.component.ts

```diff
@@ -0,0 +1,174 @@
+
+enum UrlTestState {
+  Inactive,
+  Loading,
+  Success,
+  Failure
+}
+
+@Component({
+  selector: 'app-notification-details',
+  templateUrl: './notification-details.component.html',
+  styleUrls: ['./notification-details.component.scss']
+})
+
+export class NotificationDetailsComponent implements OnInit, OnDestroy {
+  public tabValue: NotificationTabName = 'details';
+  public notification = new NotificationRule('', '', null, '', null, '', false);
+  public updateForm: FormGroup;
+  public saveInProgress = false;
+  public testInProgress = false;
+  public saveSuccessful = false;
+  public hookStatus = UrlTestState.Inactive;
+  private isDestroyed = new Subject<boolean>();
+  notificationDetailsLoading = true;
+
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<NgrxStateAtom>,
+    private layoutFacade: LayoutFacadeService
+  ) {
+
+    this.updateForm = this.fb.group({
+      // Must stay in sync with error checks in data-feed-details.component.html
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      // Note that URL here may be FQDN -or- IP!
+      targetUrl: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
+    });
+
+    this.store.select(routeParams).pipe(
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.store.dispatch(new GetNotification({ id }));
+      });
+
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(notificationFromRoute)
+    ]).pipe(
+      filter(([status, notification]) =>
+      status === EntityStatus.loadingSuccess && !isNil(notification)),
+      takeUntil(this.isDestroyed))
+      .subscribe(([_, notification]) => {
+        this.notification = notification;
+        setTimeout(() => (
+          this.updateForm.controls.name.setValue(this.notification.name),
+          this.updateForm.controls.targetUrl.setValue(this.notification.targetUrl),
+          this.notificationDetailsLoading = false
+        ),
+        200);
+      });
+
+    this.store.pipe(
+      select(updateStatus),
+      takeUntil(this.isDestroyed),
+      filter(state => this.saveInProgress && !pending(state)))
+      .subscribe((state) => {
+        this.saveInProgress = false;
+        this.saveSuccessful = (state === EntityStatus.loadingSuccess);
+        if (this.saveSuccessful) {
+          this.updateForm.markAsPristine();
+        }
+      });
+  }
+
+  public ngOnInit(): void {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+  }
+
+  public saveNotification(): void {
+    this.saveSuccessful = false;
+    this.saveInProgress = true;
+    this.notification.name = this.updateForm.controls['name'].value.trim();
+    this.notification.targetUrl = this.updateForm.controls['targetUrl'].value.trim();
+    this.store.dispatch(new UpdateNotification({
+      notification: this.notification, username: '', password: ''
```

Comment:
Why accept the username and password properties if you never use them?

Reply:
I have removed this username and password from here.